src/NUnitFramework/framework/Internal/TestProgressReporter.cs

```diff
@@ -30,6 +31,9 @@ public TestProgressReporter(ICallbackEventHandler handler)
 
         #region ITestListener Members
 
+        [ThreadStatic]
+        private static StringBuilder? _testStartedStringBuilder;
```

Comment:
This feels like a good step forward here, thanks! I don't have a great perfect model of the framework's approach to parallelization but this feels safe since we internally only use it on-stack.

I _think_ this will still create a builder per thread. This change makes me curious if this opens the door to take this further in future PRs by making a formal pooling mechanism with rent/release mechanics so it's only _at most_ a builder per thread in the worst case (ie, when all worker threads happen to be processing a `TestStarted` at the same time).

Reply:
I think the easiest approach might be just taking the source from .NET: https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Text/StringBuilderCache.cs , it's MIT licensed so should not be a problem and already battle-tested. Here I just tried to avoid bringing too much new stuff in, but I agree that general solution would be better.

Reply:
Seems that NUnit would require larger than 360 chars default as it's building XML, Jint is using a pool which is not bound to threads which might work better here: https://github.com/sebastienros/jint/blob/main/Jint/Pooling/StringBuilderPool.cs